torch/_inductor/ir.py

```diff
@@ -595,6 +595,9 @@ def num_splits(
         def _is_static(x):
             return isinstance(x, (int, sympy.Integer))
 
+        reduction_numel_hint = V.graph.sizevars.symbolic_hint(reduction_numel)
+        numel_hint = V.graph.sizevars.symbolic_hint(sympy_product(ranges))
+
```

Comment:
Maybe as a follow up, add guards based on the heuristics here: https://github.com/pytorch/pytorch/blob/27705b3b40220d8f299f607eb4dafdd77484be7a/torch/_inductor/ir.py#L624